### PR TITLE
Essentials revision

### PR DESCRIPTION
**Pull request by jamesm-macrometa**
Revised Essentials overview page and renamed it "Introduction."  Moved redundant info into appropriate quickstart sections.

@astrotars I added a new "Quickstart" file under Collections, I'm probably going to need your help properly adding it to the Docusaurus structure.

cc @diana-macrometa

### CODE REVIEW

**Comment by astrotars** [comment]
@jamesm-macrometa I pushed a couple of updates, including some tweaks to position Quickstart as the first item in the collections directory.

Some updates that were made:

1. Broken path names due to filename changes

Screen Shot 2022-06-07 at 10 14 08 AM

2. Fixed broken path to image

Screen Shot 2022-06-07 at 10 14 20 AM

3. Updated `position` in `_category_.json` files in `collections`

Screen Shot 2022-06-07 at 10 23 39 AM

**Review by astrotars — APPROVED** [review]
👍 @jamesm-macrometa

File: docs/cep/quickstart.md [hunk 1]
```diff
@@ -29,13 +31,15 @@ You can process streams to perform the following actions with your data:
 * Clean data by filtering it and by modifying the content in messages. For example, obfuscating sensitive information.
 * Derive insights by identifying event patterns in data streams.
 * Summarize data with time windows and incremental aggregations.
-* Real-time ETL for collections, tailing files, and scraping HTTP endpoints.
+* Extract, transform, and load (ETL) collections, tailing files, and scraping HTTP endpoints.
 * Integrating stream data and trigger actions based on the data. This can be a single service request or a complex enterprise integration flow.
 
-For the following examples, assume these credentials:
+# Example
```
Comment (diana-macrometa) [inline docs/cep/quickstart.md]: Why is there an h1 this far down the page? There should only be one h1, and it should be at the top of the page.

File: docs/collections/documents/quickstart.md [hunk 2]
```diff
@@ -6,47 +6,90 @@ title: Quickstart
...
+* Queries filter documents based on specified criteria, compute new data, and update or delete existing documents. Queries can be as simple as a "query by example" or as complex as "joins" using many collections or traversing graph structures. GDN queries are written in the [C8 Query Language (C8QL)](c8ql/introduction.md).
+
+* Cursors are used to iterate over the result of queries, so that you get easily processable batches instead of one big hunk.
 
-The value of the `_key` attribute can be specified by the user when creating a document. `_id` and `_key` values are immutable once the document has been created. The `_rev` value is maintained by GDN automatically.
+* Indexes can speed up your searches. Refer to the [Indexing](indexing/overview.md) section for more information.
 
 
-## Pre-requisite
+# Example
```
Comment (diana-macrometa) [inline docs/collections/documents/quickstart.md]: Why is there an h1 on line 88 here?

File: docs/collections/graphs/quickstart.md [hunk 3]
```diff
@@ -6,73 +6,27 @@ title: Quickstart
...
 * OUTBOUND: `_from` → `_to`
 * INBOUND: `_from` ← `_to`
 * ANY: `_from` ↔ `_to`.
 
-:::note
-If you are new to Macrometa GDN, we strongly recommend reading **[Essentials](../../essentials/overview.md)** of Macrometa GDN.
-:::
-
-## Pre-requisite
+# Example
```
Comment (diana-macrometa) [inline docs/collections/graphs/quickstart.md]: Another h1 in the wrong place. Needs to be h2 or above.

File: docs/collections/keyvalue/quickstart.md [hunk 4]
```diff
@@ -6,15 +6,9 @@ title: Quickstart
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 
-Macrometa GDN is a geodistributed real-time coordination-free materialized views engine that supports multiple data models. You can use GDN as a geo-replicated real-time key-value datastore or database. 
+In GDN, each document stored in a collection contains a `_key`, and the rest of the document is its *value*. The only key-value operations available are key lookups (single & batch) and key-value pair insertions and updates. If no sharding attribute is specified, `_key` is used for sharding the data instead.
```
Comment (diana-macrometa) [inline docs/collections/keyvalue/quickstart.md]:
```suggestion
In GDN, each document stored in a collection contains a `_key`, and the rest of the document is its *value*. The only key-value operations available are key lookups (single and batch) and key-value pair insertions and updates. If no sharding attribute is specified, `_key` is used for sharding the data instead.
```

File: docs/essentials/introduction.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+When you choose a database or stream processing system, you're actually choosing three technologies: storage, a data model, and an API and query language.
+
+!!! example
+    If you choose Postgres, you are choosing the Postgres storage engine, a relational data model, and the SQL query language. 
+    
+    If you choose MongoDB, you are choosing the MongoDB distributed storage engine, a document data model, and the MongoDB API. 
+    
+    In systems like these, features are integrated across all layers. For example, an index provided by a data model is automatically integrated into the storage and query language.
+
+
```
Comment (diana-macrometa) [inline docs/essentials/introduction.md]:
```suggestion
```

File: docs/essentials/introduction.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+A RestQL is set of named, parameterized C8QL queries stored in GDN that can be executed from a dedicated REST endpoint. The RestQL will be created automatically globally and is available from the region closest to the user. We recommend using RestQL to build applications backed by GDN as opposed to querying with raw C8QL directly from application code or setting up a centralized middleware.
+
+RestQLs can be created and updated using the GDN Console or by using the REST API directly. Each RestQL is tied to a specific query text and parameter set. You can set default values for query parameters (making them optional during executions of your RestQL), or you can make them mandatory for each execution (failing to pass along will result in an error).
+
+Each RestQL is exposed as its own endpoint and is protected. The RestQLs are organized by the GeoFabric (or database) enabling you to have different RestQLs for different geo-regions as well as for different fabrics within same region.
+
+For more information, refer to the [C8 Query Language](../c8ql/introduction.md) section.
+
+## Sharding & Replication
```
Comment (diana-macrometa) [inline docs/essentials/introduction.md]:
```suggestion
## Sharding and Replication
```

Comment (diana-macrometa) [reply docs/essentials/introduction.md]: Avoid stacked headings. Need a paragraph of plain text between these two.

File: docs/essentials/introduction.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+
+This allows to scale down a GDN cluster without service interruption, loss of fault tolerance or data loss. Furthermore, one can re-balance the distribution of the shards, either manually or automatically.
+
+Similarly when messages are produced on a GDN stream, they are first persisted in the local datacenter and then forwarded asynchronously to the remote datacenters.
+
+In normal cases, when there are no connectivity issues, messages are replicated immediately, at the same time as they are dispatched to consumers of local datacenter. Typically, end-to-end delivery latency is defined by the network round-trip time (RTT) between the remote regions.
+
+Applications can create producers and consumers in any of the datacenters, even when the remote datacenters are not reachable (like during a network partition).
+
+> Subscriptions are local to a GDN datacenter.
```
Comment (diana-macrometa) [inline docs/essentials/introduction.md]: What is this? Is it intended to be a note?

File: docs/streams/quickstart.md [hunk 8]
```diff
@@ -6,23 +6,29 @@ title: Quickstart
...
+A stream is a named channel for sending messages. Each stream is backed by a `distributed append-only` log and can be `local` (at one edge location only) or `global` (across all edge locations in the Fabric).
 
-GDN streams are a high-performance solution for server-to-server messaging. It provides:
+Messages from publishers are only stored once on a stream, and can be consumed as many times as necessary by consumers. The stream is the source of truth for consumption. Although messages are only stored once on the stream, there can be different ways of consuming these messages.
+
+Consumers are grouped together for consuming messages. Each group of consumers is a subscription on a stream. Each consumer group can have its own way of consuming the messages: exclusively, shared, or failover.
+
+Streams provide:
 
 - Seamless geo-replication of messages across regions
```
Comment (diana-macrometa) [inline docs/streams/quickstart.md:25]: James, we're going to need to standardize how we format things like lists and italics. Something for the style guide.

File: docs/streams/quickstart.md [hunk 9]
```diff
@@ -38,14 +44,12 @@ Messages are the basic "unit" of Streams. They're what producers publish to stre
 | Publish Time         | The timestamp of when the message was published (automatically applied by the producer) |
 | Event Time           | An optional timestamp that applications can attach to the message representing when something happened, e.g. when the message was processed. The event time of a message is 0 if none is explicitly set. |
 
-:::note
-If you are new to Macrometa GDN, we strongly recommend reading **[Essentials](../essentials/overview.md)** of Macrometa GDN.
-:::
-## Pre-requisite
 
-Let's assume your
+# Example
```
Comment (diana-macrometa) [inline docs/streams/quickstart.md]: Fix this h1.

**Review by diana-macrometa — CHANGES_REQUESTED** [review]
Looks good, needs a few changes.

**Review by diana-macrometa — APPROVED** [review]
This PR is pretty large. I suggest we merge it and tackle the rest in separate PRs.

File: docs/collections/graphs/quickstart.md [hunk 10]
```diff
@@ -6,76 +6,30 @@ title: Quickstart
...
 
-Let's assume your
+For this example, assume the following credentials:
 
-* Tenant name is `nemo@nautilus.com` and 
+* Tenant name is `nemo@nautilus.com`.
 * User password is `xxxxxx`.
 
-## Driver download
+### Driver download
```
Comment (dustinlarimer) [inline docs/collections/graphs/quickstart.md]: Should this be title case, match other H2s?

File: docs/essentials/introduction.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+|a | b | not ok|
+|a | a, b |	ok |
+|a, b| a | not ok|
+|a, b| b | not ok|
+|a, b| a, b| ok|
+|a, b| a, b, c| ok|
+|a, b, c| a, b |not ok|
+|a, b, c| a, b, c| ok|
+
+### Replication Within Datacenter
```
Comment (dustinlarimer) [inline docs/essentials/introduction.md:99]: We should shore up "data center" vs. "datacenter". It appears we use "data center" somewhat consistently on product/landing pages of the website, and "datacenter" here and in the blog.

File: docs/essentials/introduction.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+
+If an instance that holds a leader copy of a shard fails, then the leader can no longer serve any requests. After 15 seconds of missed heartbeats, the supervision process promotes another shard to become the leader. The other surviving replicas automatically synchronize their data with the new leader. When the instance with the original leader comes online, it becomes a follower and synchronizes its data with the leader. Both leader and follower shards can be moved between instances without service interruptions.
+
+Applications can create producers and consumers in any shard. If follower shards are not reachable, the producers and consumers are automatically synchronized when the follower shard comes back online. Ordering is still guaranteed on a per-producer basis. However, subscriptions are local so you must recreate the subscription in the desired datacenter.
+
+### Replication Across Datacenters
```
Comment (dustinlarimer) [inline docs/essentials/introduction.md:111]: Another "datacenter" vs. "data center"

File: docs/streams/quickstart.md [hunk 13]
```diff
@@ -693,7 +697,7 @@ var producer;
 </TabItem>
 </Tabs>  
 
-## Pub-Sub with streams in browser
+### Pub-Sub with streams in browser
```
Comment (dustinlarimer) [inline docs/streams/quickstart.md]: Sentence-case - switch to title case?

File: docs/essentials/introduction.md (hunk 11 above)
Comment (jamesm-macrometa) [reply docs/essentials/introduction.md:99]: @diana-macrometa We should add a note about this in the style guide if we don't already have one.  Do you prefer "datacenter" or "data center"?

**Pull request merged**